TesseractOCR/G8Tesseract.mm

```diff
@@ -287,18 +307,28 @@ - (void)setVariableValue:(NSString *)value forKey:(NSString *)key
      * _tesseract->SetVariable("language_model_penalty_non_freq_dict_word", "0");
      * _tesseract->SetVariable("language_model_penalty_non_dict_word ", "0");
      */
-
+    
     [self resetFlags];
 
+    if (!value) {
```

Comment:
I thought about it, but I chose the native iOS behavior, which is crashing if key is nil.
Just try `setValue:forKey:` with a nil key to any object. 
Also in xCode 7 function parameters are now can be nullable and nonnull. So if we later specify none keyword for the key parameter the static analyzer will not allow the user to send nil there. I'll do this later in the xCode 7 branch.